File: trunk/plsi_camp/bcast.f90
program bcast
include 'mpif.h'

integer i, ierr, rank, size, imsg(4)

call MPI_INIT(ierr)
call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierr)
call MPI_COMM_SIZE(MPI_COMM_WORLD, size, ierr)

if (rank == 0) then
    do i = 1, 4
    imsg(i) = i
    enddo
else
    do i = 1, 4
    imsg(i) = 0
    enddo
endif

print *, 'rank:', rank, ', before:', imsg

call MPI_BCAST(imsg, 4, MPI_INTEGER, 0, MPI_COMM_WORLD, ierr)

print *, 'rank:', rank, ', after: ', imsg

call MPI_FINALIZE(ierr)

end

File: trunk/plsi_camp/bcast.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char *argv[])
{
    int myrank, i;
    int imsg[4];

    MPI_Init(&argc, &argv);
    MPI_Comm_rank(MPI_COMM_WORLD, &myrank);

    if (myrank == 0) for (i = 0; i < 4; i++) imsg[i] = i + 1;
    else for (i = 0; i < 4; i++) imsg[i] = 0;

    printf("%d: BEFORE:", myrank);
    for (i = 0; i < 4; i++) printf("%d", imsg[i]);
    printf("\n");

    MPI_Bcast(imsg, 4, MPI_INT, 0, MPI_COMM_WORLD);
    
    printf("%d: AFTER:", myrank);
    for (i = 0; i < 4; i++) printf("%d", imsg[i]);
    printf("\n");

    MPI_Finalize();

    return 0;
}
